File: examples/01_TTS/python/tts_flite.py
# Example TTS - "flite"

# To install flite please see Tech Resources:
# https://github.com/juxtapix/OkRobotReboot/wiki/02.-TTS#technical-resources

# import "os" to access operating system functionalities
import os
# import "sys" to access the interpreter functionalities
import sys

def say(something):
    command = "flite \"{0}\" -o test.wav && afplay test.wav".format(something)
    # command = "flite -voice voices/cmu_us_jmk.flitevox \"{0}\" -o test.wav && afplay test.wav".format(something)
    os.system(command)

 # "sys.argv" returns an array with the arguments passed when Python interpreter was launched.
say(sys.argv[1])

File: examples/01_TTS/node/tts_flite.js
// Example TTS - "flite"

// To install "flite" please see Tech Resources:
// https://github.com/juxtapix/OkRobotReboot/wiki/02.-TTS#technical-resources

// require "child_process" to run Child Process Applications
var exec = require('child_process').execSync;
// require "util" to access utility functions such as "format"
var util = require('util');


function say(something){
  command = util.format('flite "%s" -o test.wav && afplay test.wav', something);
  // command = util.format('flite -voice voices/cmu_us_jmk.flitevox "%s" -o test.wav && afplay test.wav', something);
  exec(command);
}

// "process.argv" returns an array with the arguments passed when Node.js process was launched.
say(process.argv[2]);
